uaclient/tests/test_apt.py

```diff
@@ -347,140 +347,190 @@ def test_errors_on_apt_helper_process_timeout(
 
 
 class TestAddAuthAptRepo:
+    @pytest.mark.parametrize("series", ("xenial", "noble"))
     @mock.patch("uaclient.apt.gpg.export_gpg_key")
     @mock.patch("uaclient.system.subp")
     @mock.patch("uaclient.apt.get_apt_auth_file_from_apt_config")
     @mock.patch("uaclient.apt.assert_valid_apt_credentials")
-    @mock.patch(
-        "uaclient.system.get_release_info",
-        return_value=mock.MagicMock(series="xenial"),
-    )
+    @mock.patch("uaclient.system.get_release_info")
     def test_add_auth_apt_repo_writes_sources_file(
         self,
         m_get_release_info,
         m_valid_creds,
         m_get_apt_auth_file,
         m_subp,
         m_gpg_export,
+        series,
         tmpdir,
     ):
         """Write a properly configured sources file to repo_filename."""
         repo_file = tmpdir.join("repo.conf").strpath
         auth_file = tmpdir.join("auth.conf").strpath
         m_get_apt_auth_file.return_value = auth_file
         m_subp.return_value = "500 esm.canonical.com...", ""  # apt policy
+        m_get_release_info.return_value = mock.MagicMock(series=series)
 
         add_auth_apt_repo(
             repo_filename=repo_file,
             repo_url="http://fakerepo",
             credentials="mycreds",
-            suites=("xenial",),
+            suites=(series,),
             keyring_file="keyring",
         )
 
-        expected_content = (
-            "deb http://fakerepo xenial main\n"
-            "# deb-src http://fakerepo xenial main\n"
-        )
+        if series == "xenial":
+            expected_content = (
+                "deb http://fakerepo xenial main\n"
+                "# deb-src http://fakerepo xenial main\n"
+            )
+            src_keyfile = os.path.join(KEYRINGS_DIR, "keyring")
+            dest_keyfile = os.path.join(APT_KEYS_DIR, "keyring")
+            gpg_export_calls = [mock.call(src_keyfile, dest_keyfile)]
+        else:
+            expected_content = (
+                "# Written by ubuntu-advantage-tools\n"
+                "\n"
+                "Types: deb\n"
+                "URIs: http://fakerepo\n"
+                "Suites: noble\n"
```

Comment:
Also best to use `{series}` instead of the hardcoded `noble`, assuming also you took the suggestion of using `SERIES_NOT_USING_DEB822` from above.